### DISCUSSION TITLE
How to run multiple environments at once?

### DISCUSSION THREAD

**Discussion by Maarten-Dekker** [serversideup/spin#98] [Q&A] [answered]
For the formatting of my projects, I use Prettier with the [prettier-plugin-blade](https://stillat.com/blade-parser/v1/installing-prettier-plugin) plugin. `useLaravelPint` is enabled in `.blade.format.json`. To format the files, we could run:

`spin run node npx prettier --write resources/`

Right now, this is returned:

```
resources/views/welcome.blade.php/bin/sh: 1: php: not found
resources/views/welcome.blade.php 163ms (unchanged)
...
```

This is expected, as we only run the Node environment.

I thought it might be possible with `spin exec node npx prettier --write resources/`, but that returns `service "node" is not running`.

What's the best approach to enable PHP? I feel that somehow I could combine both environments in one command, or do i need to create a new Dockerfile that combines both in one file?

**Comment by jaydrogers** [comment, marked as answer]
That's a bummer it requires both PHP and Node, but I get why they are doing that.

In this case, the best way to move forward would be installing node within the PHP container. Don't do this for your production image though (I assume you have a build step for compiling assets).

Instead, you can use Multi-stage builds to do it for development only: https://docs.docker.com/build/building/multi-stage/

Hope this helps!
